### ISSUE TITLE
accuracy函数报错

### ISSUE THREAD

**Issue by betterpig** [PaddlePaddle/Paddle#34340]
- 版本、环境信息：
    1）PaddlePaddle版本：release/2.1
    2）CPU：Intel(R) Xeon(R) CPU E5-2620 v2 @ 2.10GHz
    3）GPU：Tesla K40m ,CUDA10.2 ,CUDNN7.6
    4）系统环境：请您描述系统类型、版本（如Mac OS 10.14），Python版本
- 训练信息
    1）单机，单卡
    2）Operator信息：paddle.metric.accuracy()
- 复现信息：直接运行

- 问题描述：同样的参数，能够计算loss，但是计算acc时报错，按照报错信息修改格式后，又报错
修改前说不支持float
![image](https://user-images.githubusercontent.com/51314274/126735025-4f6a89a4-4d1b-49dd-8f1e-391b9e2b5b65.png)
修改后又说不行
![image](https://user-images.githubusercontent.com/51314274/126734914-872907e4-1d67-4563-bb45-bf01b70719d3.png)

- 代码

```
import paddle
from paddle.nn import Linear
import paddle.nn.functional as F
import os
import numpy as np
import matplotlib.pyplot as plt

paddle.vision.set_image_backend('cv2')
def norm_img(img):
    assert len(img.shape) == 3
    batch_size,img_h,img_w = img.shape[0],img.shape[1],img.shape[2]
    img = img/255
    img = paddle.reshape(img,[batch_size,img_h*img_w])
    return img
class MNIST(paddle.nn.Layer):
    def __init__(self):
        super(MNIST, self).__init__()
        self.fc = paddle.nn.Linear(in_features=784, out_features=1)
    def forward(self, inputs):
        outputs = self.fc(inputs)
        return outputs
def train(model):
    model.train()
    train_loader = paddle.io.DataLoader(paddle.vision.datasets.MNIST(mode="train"),
                                        batch_size=16,
                                        shuffle=True)
    opt= paddle.optimizer.SGD(learning_rate=0.001,parameters=model.parameters())
    EPOCH_NUM = 10
    for epoch in range(EPOCH_NUM):
        for batch_id, data in enumerate(train_loader()):
            images = norm_img(data[0]).astype('float32')
            labels = data[1].astype('float32')
            predicts = model(images)
            loss = F.square_error_cost(predicts, labels)
            avg_loss = paddle.mean(loss)
            if batch_id % 1000 == 0:
                print("epoch_id: {}, batch_id: {}, loss is: {}".format(epoch, batch_id, avg_loss.numpy()))
            avg_loss.backward()
            opt.step()
            opt.clear_grad()
model = MNIST()
train_dataset = paddle.vision.datasets.MNIST(mode='train')
if os.path.exists('mnist.paparams'):
    train(model)
    paddle.save(model.state_dict(), './mnist.pdparams')
else:
    param_dict = paddle.load('mnist.pdparams')
    model.load_dict(param_dict)
test_dataset = paddle.vision.datasets.MNIST(mode='test') 
test_loader = paddle.io.DataLoader(test_dataset,batch_size=64, drop_last=True)
model.eval()
for batch_id, data in enumerate(test_loader()):
    x_data = norm_img(data[0]).astype('float32')
    y_data = data[1].astype('float32')
    predicts = model(x_data)   
    loss = paddle.nn.CrossEntropyLoss(predicts, y_data)
    acc = paddle.metric.accuracy(predicts, y_data)
    if (batch_id+1) % 30 == 0:
        print("batch_id: {}, loss is: {}, acc is: {}".format(batch_id+1, loss.numpy(), acc.numpy()))
```

**Comment by betterpig** [comment]
环境：docker-ubuntu16.04; python3.6;

**Comment by ceci3** [comment]
您好，y_data需要是int64_t类型的，可以看下文档里写的参数类型说明。https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/metric/accuracy_cn.html#accuracy

**Comment by betterpig** [comment]
> 您好，y_data需要是int64_t类型的，可以看下文档里写的参数类型说明。https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/metric/accuracy_cn.html#accuracy

问题已解决，感谢！是我看错了参数说明，第一个是y_predict，可以是float；第二个是y_data，要求是int64。
